Add packaging info and scripts

StefansM opened the pull request:
This commit adds some documentation of the packaging process, some git hooks for safety and an autoupdate script for trivial version bumps.

StefansM committed:
Add packaging info and scripts

livibetter commented on README.md:
```diff
@@ -23,7 +23,39 @@ See [@Pipeseroni/Packagers][packagers] for list of members who are responsible f
+
+### Automated packaging
+
+The entire packaging process can be automated with [scripts/autoupdate], which
```
You should use [relative link](https://blog.github.com/2013-01-31-relative-links-in-markup-files/), I believe the following would work:

```markdown
... with [scripts/autoupdate](scripts/autoupdate), which
```

livibetter replied:
You forgot this one.

livibetter commented on README.md:
```diff
@@ -23,7 +23,39 @@ See [@Pipeseroni/Packagers][packagers] for list of members who are responsible f
+
+### Git hooks
+
+It is recommended that you install [hooks/pre-commit] as a pre-commit hook (`cp
```
Did you mean `scripts/...` here and in two more occurrences?

livibetter commented on scripts/pre-commit:
```diff
@@ -0,0 +1,34 @@
+
+# Check if arrays are equal. They must be the same length and contain equal
+# elements.
+[[ ${#pkgbuild_hashes[@]} == ${#actual_hashes[@]} ]] || hash_mismatch
```
What is this `hash_mismatch`?  Or is this just intentionally to have a "command not found" ?

Also I think a simple:

```bash
[[ "${pkgbuild_hashes[@]}" == "${actual_hashes[@]}" ]] || hash_mismatch
```

should suffice the comparison without actually checking the length and comparing each hash individually.

StefansM committed:
Fix relative links to scripts

StefansM committed:
Remove obsolete code and actually exit on error

The hash-checking code removed in this commit is made useless by the
call to `diff`, so we can remove it.

StefansM commented:
Wow, embarrassing. Yet more proof code reviews are a good thing. Thanks!

Also, at the top of `pre-commit` is an `exit 0`.

I will fix the scripts, test them more thoroughly, and update this PR.

StefansM committed:
Add check for branch in pre-commit

StefansM committed:
Add CC0 waiver to scripts

StefansM committed:
Add CC0 license for packaging repository.

livibetter reviewed (changes requested):
Also update the Copyright section in README, and require all contribution to be under the same term, you might want to add another Contributing section.

livibetter committed:
fix relative link to scripts/autoupdate

livibetter commented:
@StefansM I'm merging this and the other two, because they were (almost) ready to be merged.  I hope I am not overstepping or interrupt what you still have any commits planned for them.
